app/src/main/java/com/group80/uoftinder/logout/LogOutInteractor.java

```diff
@@ -0,0 +1,19 @@
+package com.group80.uoftinder.logout;
+
+import com.google.firebase.auth.FirebaseAuth;
+
+public class LogOutInteractor {
+    final LogOutPresenter logOutPresenter;
+
+    public LogOutInteractor(LogOutPresenter logOutPresenter) {
+        this.logOutPresenter = logOutPresenter;
+    }
+
+    /**
+     * Signs out the user, switches UI back to LoginActivity
+     */
+    public void signOut() {
+        FirebaseAuth.getInstance().signOut();
+        logOutPresenter.showLogin();
```

Comment:
This violates Clean Architecture as a use case is not able to call the presenter. There must be an interface, LogOutPresenterInterface, that the LogOutPresenter implements. Then, the use case can be passed in the LogOutPresenter as a LogOutPresenterInterface type and ``signOut`` can make the call: ``logOutPresenterInterface.showLogin``. An alternative is that you can somehow get the presenter's signOut method to have a return value. The Presenter will then interpret this return value to perform a certain action